Sim/SimG4Fast/src/components/SimG4ParticleSmearFormula.h

```diff
@@ -44,26 +46,40 @@ class SimG4ParticleSmearFormula: public GaudiTool, virtual public ISimG4Particle
    *   @return status code
    */
   virtual StatusCode smearMomentum(CLHEP::Hep3Vector& aMom, int aPdg = 0) final;
-
-  /**  Smear the energy of the particle
-   *   @param aEn Particle energy to be smeared.
-   *   @param[in] aPdg Particle PDG code.
-   *   @return status code
+  /**  Get the names of the volumes where fast simulation should be performed.
+   *   @return vector of volume names
+   */
+  inline virtual const std::vector<std::string>* volumeNames() const final {return &m_volumeNames;};
+  /**  Get the minimum momentum that triggers fast simulation
+   *   @return maximum p
+   */
+  inline virtual double minP() const final {return m_minP;};
+  /**  Get the maximum momentum that triggers fast simulation
+   *   @return maximum p
+   */
+  inline virtual double maxP() const final {return m_maxP;};
+  /**  Get the maximum pseudorapidity that triggers fast simulation
+   *   @return maximum p
    */
-  virtual StatusCode smearEnergy(double& aEn, int aPdg = 0) final;
+  inline virtual double maxEta() const final {return m_maxEta;};
+
 private:
-  /// string defining a TFormula representing resolution energy-dependent for the smearing (set by job options)
-  std::string m_resolutionEnergyStr;
-  /// string defining a TFormula representing resolution momentum-dependent for the smearing (set by job options)
-  std::string m_resolutionMomentumStr;
-  /// TFormula representing resolution energy-dependent for the smearing
-  std::unique_ptr<TFormula> m_resolutionEnergy;
   /// TFormula representing resolution momentum-dependent for the smearing
-  std::unique_ptr<TFormula> m_resolutionMomentum;
+  TFormula m_resolutionMomentum;
   /// Random Number Service
   SmartIF<IRndmGenSvc> m_randSvc;
   /// Gaussian random number generator used for smearing with a constant resolution (m_sigma)
   IRndmGen* m_gauss;
+  /// Names of the parametrised volumes (set by job options)
+  std::vector<std::string> m_volumeNames;
+  /// string defining a TFormula representing resolution momentum-dependent for the smearing (set by job options)
+  std::string m_resolutionMomentumStr;
```

Comment:
I updated the md documentation and added a doxygen reference to it in the description above.